src/main/java/de/uni_stuttgart/it_rex/course/service/written/ProgressTrackingService.java

```diff
@@ -0,0 +1,284 @@
+
+    /**
+     * Tracker mapper.
+     */
+    private final CourseProgressTrackerMapper courseProgressTrackerMapper;
+
+    /**
+     * Content tracker repository.
+     */
+    private final ContentProgressTrackerRepository
+        contentProgressTrackerRepository;
+
+    /**
+     * Content tracker mapper.
+     */
+    private final ContentProgressTrackerMapper contentProgressTrackerMapper;
+
+    /**
+     * Required in order to attach content progress trackers to existing content references only.
+     */
+    private final ContentReferenceRepository contentReferenceRepository;
+
+    /**
+     * Constructor.
+     *
+     * @param courseProgressTrackerRepository
+     * @param courseProgressTrackerMapper
+     * @param contentProgressTrackerRepository
+     */
+    @Autowired
+    public ProgressTrackingService(
+        final CourseProgressTrackerRepository courseProgressTrackerRepository,
+        final CourseProgressTrackerMapper courseProgressTrackerMapper,
+        final ContentProgressTrackerRepository contentProgressTrackerRepository,
+        final ContentProgressTrackerMapper contentProgressTrackerMapper,
+        final ContentReferenceRepository contentReferenceRepository) {
+        this.courseProgressTrackerRepository = courseProgressTrackerRepository;
+        this.courseProgressTrackerMapper = courseProgressTrackerMapper;
+        this.contentProgressTrackerRepository =
+            contentProgressTrackerRepository;
+        this.contentProgressTrackerMapper = contentProgressTrackerMapper;
+        this.contentReferenceRepository = contentReferenceRepository;
+    }
+
+    /**
+     * Initialize the content progress tracking for a user regarding one specific content reference.
+     *
+     * To be used only once per user to create a tracker for the content reference.
+     *
+     * As all ContentProgressTrackers logically belong to one @{@link CourseProgressTracker} as all contents
+     * belong to a course, from a business logic perspective, the relationship is already implicitly defined.
+     * However, for fast access from a @{@link CourseProgressTracker} to all its ContentProgressTrackers, we
+     * explicitly store this relationship.
+     *
+     * @param contentReferenceDTO the content reference to track - must be valid
+     * @param userId the user to start the tracking for
+     * @param courseProgressTrackerId the ID of the course progress tracker that the content reference belongs to,
+     *                                must be a valid and existing ID
+     * @return the created {@link de.uni_stuttgart.it_rex.course.service.dto.written_dtos.ContentReferenceDTO}
+     *
+     * @throws NotFoundException if either the content reference or the course progress tracker do not exist
+     */
+    @Transactional
+    public ContentProgressTrackerDTO startContentProgressTracking(final ContentReferenceDTO contentReferenceDTO, final UUID userId, final UUID courseProgressTrackerId){
```

Comment:
In der Methode ziehen wir nur die ID aus dem contentReferenceDTO, und der Endpunkt nimmt entsprechend das ganze DTO entgegen vom Frontend, um den Service hier füttern zu können. Hätte man da nicht gleich einfach nur die ID als Parameter nehmen können? Ist irgendwie ein wenig inkonsistent und unnötig kompliziert finde ich, aber vielleicht übersehe ich was.